MobileCenter/MobileCenter/MSCustomProperties.m

```diff
@@ -59,10 +58,32 @@ - (BOOL) isValidKey:(NSString *)key {
     MSLogError([MSMobileCenter logTag], @"Custom property \"%@\" must match \"%@\"", key, kKeyPattern);
     return NO;
   }
+  if (key.length > maxPropertyKeyLength) {
+    MSLogError([MSMobileCenter logTag], @"Custom property \"%@\" length cannot be longer than \"%d\" characters.", key, maxPropertyKeyLength);
+    return NO;
+  }
   if ([self.properties objectForKey:key]) {
     MSLogWarning([MSMobileCenter logTag], @"Custom property \"%@\" is already set or cleared and will be overridden.", key);
+  } else if ([self properties].count >= maxPropertiesCount) {
+    MSLogError([MSMobileCenter logTag], @"Custom properties cannot contain more than \"%d\" items.", maxPropertiesCount);
+    return NO;
   }
   return YES;
 }
 
+- (BOOL) isValidValue:(NSObject *)value {
+  if (value) {
+    if ([value isKindOfClass:[NSString class]]) {
```

Comment:
You should turn it to string and validate it as a string because the limitation is on the number of characters.